content/en/docs/contributing/coding-conventions.md

```diff
@@ -0,0 +1,823 @@
+
+```c
+switch (var) {
+case 0:
+    /* … */
+    break;
+case 1:
+    /* … */
+    break;
+default:
+    /* … */
+    break;
+}
+```
+
+### Goto Statements
+
+Goto labels should not **be indented**.
+The name of the label should be: `err_…` or `out_…`, depending on the context.
+
+Generally, goto should be avoided for simple “retry”-style constructs.
+    Use a while-loop instead.
+
+Gotos can be a good solution to implement a common error/finalization path.
+This reduces the probability that some error paths may leak resources or do not release all locks (in the correct order).
+This can easily happen as error paths are usually less well tested. If, however, possible, return directly.
+
+```c
+int cc_myfunc(void)
+{
+    void *a, *b;
+    int rc = 0;
+
+    a = malloc(...);
+    if (unlikely(!a)) {
+            rc = -ENOMEM;
+            goto err_out;
+    }
+
+    b = malloc(...);
+    if (unlikely(!b)) {
+            rc = -ENOMEM;
+            goto err_free_a;
+    }
+
+    /* ... */
+    free(b);
+
+err_free_a:
+        free(a);
+err_out:
+        return rc;
+}
+✅
+```
+
+```c
+int cc_myfunc(void)
+{
+    void *a, *b;
+
+    a = malloc(...);
+    if (unlikely(!a))
+            return -ENOMEM;
+
+
+    b = malloc(...);
+    if (unlikely(!b)) {
+            free(a);
+            return -ENOMEM;
+    }
+
+    /* ... */
+    free(b);
+    free(a);
+    return 0;
+}
+❌
+```
+
+### Assembly
+
+#### Code
+
+Indentation for ASM code:
+
+* Multi line (tab at the beginning, tab between instr. and operands, `\n` at the end)
+* Could also remove tab but that means `\n\t` at the end, which is not nice, if at some point code is changed and label has to be added.
+
+#### Inline Assembly
+
+```c
+__asm__ __volatile__ (
+    “1:      instr      op, op\n”
+    “        instr2     op, op\n”
+    :::);
+```
+
+* Single-line
+* With / without label
+* `__asm__ __volatile__` vs. `asm volatile`
+* Aim to add a comment for each inline assembly line, detailing what it does
+    * Ignore for repetitive operations (such as mov-s)
+    * Ignore long lines for inline assembly, when including comments
+    * Aim to have short comments
+    We could also have multi-line comments
+
+### Breaking Long Lines
+
+TODO: Two questions:
+
+* When do we break?
+When reaching 80 characters?
+* Where do we break?
+* How do we align after the break?
+
+1. Proposal MR: Break before function name (after function return type)
+1. Proposal MR: Align function arguments after break to open parenthesis
+1. Decision: agreed
+
+TODO: How do we align when we break on conditions, for loops?
+
+For functions, align arguments on different line:
+
+```c
+UK_SYSCALL_R_DEFINE(int, getpeername, int, sock,
+»       »           struct sockaddr *restrict, addr,
+»       »           socklen_t *restrict, addr_len)
+```
+
+For conditions, always aligned to the parenthesis.
+To be voted via chat.
+One breaking rule to rule them all.
+TODO: Wait for Michalis’s insight.
```

Comment:
Update according to [this](https://discord.com/channels/762976922531528725/857555406187528212/1110131214393806860).